### PR TITLE
DPC-1612 Split DPC API into public and admin servers

### PR DESCRIPTION
**Pull request by MrMorie**
### Fixes [DPC-1612](https://jira.cms.gov/browse/DPC-1612)

The DPC v2 API has two different sets of users: 1) our external users, who are getting claims data, and 2) our web applications, which are performing administrative functions or operating on behalf of many authenticated external users.

For clarity and to provide a network layer of protection, we would like to host the endpoints for these two groups of users on separate ports, represented in code by separate routers
### Proposed Changes
- Split the DPC API router so that administrative endpoints are hosted from a separate port.

### Change Details
- Created public and admin packages
- Created a new server struct to hold common server code
- Created admin and public routers
- Updated dockerfiles
- Updated docker compose files
- Kept the public server running on port 3000 (original port used by API)
- Exposed admin server on port 3011 (can be changed if needed)
- Split router unit tests
- Fixed DPC capabilities bug that was found during testing

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation
Tested locally

Note: 
3000 port is PUBLIC server
3011 port is ADMIN server
Screen Shot 2021-07-26 at 9 36 26 AM
Screen Shot 2021-07-26 at 9 37 25 AM
Screen Shot 2021-07-26 at 9 37 12 AM
Screen Shot 2021-07-26 at 9 36 40 AM

Screen Shot 2021-07-25 at 3 05 37 PM
Screen Shot 2021-07-25 at 7 31 32 PM

Ignore the 400 status code, the point is that it made it to the controller (route existed)
Screen Shot 2021-07-25 at 7 31 09 PM
Screen Shot 2021-07-25 at 10 32 24 PM
Screen Shot 2021-07-25 at 10 32 07 PM

### Feedback Requested
Are we ok with port 3011?
Should v2/Organization/:id be available on the public server

### CODE REVIEW

File: dpc-go/dpc-api/src/v2/implementer_test.go [hunk 1]
```diff
@@ -73,13 +73,13 @@ func (suite *ImplementerControllerTestSuite) TestCreateImplementerBadJson() {
 func (suite *ImplementerControllerTestSuite) TestCreateImplementer() {
     //Mock impl creation
     createImplResp := ImplementerResource{}
-    faker.FakeData(&createImplResp)
+    _ = faker.FakeData(&createImplResp)
```
Comment (MrMorie) [inline dpc-go/dpc-api/src/v2/implementer_test.go:76]: Just our go linter was complaining about this.

File: dpc-go/dpc-api/src/service/public/router.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		r.Get("/metadata", mc.Read)
+
+		//TODO Tech spec mentions this route is only for the admin server, double check that its accurate.
+		//r.Route("/Organization", func(r chi.Router) {
+		//	r.Route("/{organizationID}", func(r chi.Router) {
+		//		r.Use(middleware2.OrganizationCtx)
+		//		r.With(middleware2.FHIRModel).Get("/", rc.Org.Read)
+		//	})
+		//})
+		r.Route("/Group", func(r chi.Router) {
```
Comment (MrMorie) [inline dpc-go/dpc-api/src/service/public/router.go:32]: Currently in V1 we allow the organizations to retrieve their own org via /Organization/:id (as long as the id is their own), or using /Organization where the result is a size 1 array (with their org).

Not sure if we are not going to support that any more or if the tech spec needs to be updated.

File: dpc-go/dpc-api/src/service/server.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+func NewServer(name string, port int, securityDisabled bool, handler http.Handler) *Server {
+	s := Server{}
+	s.name = name
+	s.port = port
+	s.securityDisabled = securityDisabled
+	s.server = http.Server{
+		Handler: handler,
+		Addr:    fmt.Sprintf("%s%d", ":", s.port),
+		//TODO make these configurable
+		ReadTimeout:  time.Duration(10) * time.Second,
```
Comment (MrMorie) [inline dpc-go/dpc-api/src/service/server.go:34]: I will update this.

File: dpc-go/dpc-api/src/service/admin/router.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	r.Use(middleware2.Logging())
+	r.Use(middleware2.RequestIPCtx)
+	r.With(middleware2.Sanitize).Route("/v2", func(r chi.Router) {
+		r.Use(middleware.SetHeader("Content-Type", "application/fhir+json; charset=UTF-8"))
+
+		//ORGANIZATION Routes
+		r.Route("/Organization", func(r chi.Router) {
+			r.Route("/{organizationID}", func(r chi.Router) {
+				r.Use(middleware2.OrganizationCtx)
+				//TODO Shouldn't org/:id be public?
```
Comment (MrMorie) [inline dpc-go/dpc-api/src/service/admin/router.go:26]: see my comment on the public router.

File: dpc-go/dpc-api/src/service/public/router.go (hunk 2 above)
Comment (dhgreene) [reply dpc-go/dpc-api/src/service/public/router.go:32]: Good call: yes, we should allow public access to `GET /api/v2/Organization` and/or `GET api/v2/Organization/:id`

Comment (MrMorie) [reply dpc-go/dpc-api/src/service/public/router.go:31]: actually, it looks like we don't have /api/v2/Organization in v2. Fixed /Organization/:id

Comment (MrMorie) [reply dpc-go/dpc-api/src/service/public/router.go:34]: Talked about it during standup, a future ticket will be created to add GET /Organization if needed.

**Review by charmcitygavin — APPROVED** [review]

**Review by MrBilnon — APPROVED** [review]

**Pull request merged**